### ISSUE TITLE
Click link from outside FireFox opens either two tabs (assigned container for domain) or default container (domain not assigned to a container)

### ISSUE THREAD

**Issue by terrizgit** [mozilla/multi-account-containers#1354]
- Is "Firefox will: Never remember history" in the Firefox Preferences/Options under "Privacy & Security > History" selected? Yes/No: No
- Are you using Firefox in a Private Window? Yes/No: No
- Can you see a grayed out but ticked Checkbox with the description "Enable Container Tabs" in the Firefox Preferences/Options under "Tabs"? Yes/No: Yes
- Multi-Account Containers Version: 6.1.0
- Operating System + Version: MacOs Mojave
- Firefox Version: 65.0.1
- Other installed Add-ons + Version + Enabled/Disabled-Status: 
Auto Tab Discard	0.3.1	true	{c2c003ee-bd69-42a2-b0e9-6f34222cb046}
Conex	0.9.7	true	{ec9d70ea-0229-49c0-bbf7-0df9bbccde35}
EverSync - Sync bookmarks, backup favorites	18.5.5	true	fvdmedia@gmail.com
F.B Purity - Cleans up Facebook (WX)	27.2.0.0	true	fbpElectroWebExt@fbpurity.com
Firefox Multi-Account Containers	6.1.0	true	@testpilot-containers
Firefox Pioneer	1.0	true	pioneer-opt-in@mozilla.org
Font Finder	0.3.1	true	{a658a273-612e-489e-b4f1-5344e672f4f5}
LastPass: Free Password Manager	4.25.0.4	true	support@lastpass.com
uBlock Origin	1.18.4	true	uBlock0@raymondhill.net
YouTube Flash Video Player	64.0.0	true	{f3bd3dd2-2888-44c5-91a2-2caeb33fb898}
Vertical Tabs Reloaded	0.12.1	false	verticaltabsreloaded@go-dev.de

### Actual behavior

Link clicked outside FF to a domain **not assigned** to a container: opens (1 tab) in **DEFAULT** container even if another container was in use.

### Expected behavior

Link clicked from outside FF to domain **not assigned** to a container opens new tab in **CURRENT** container. (This is similar to what would happen if I wasn't using MAC - it would open in the current window vs. opening a new window - because I have the "any link opened outside of Firefox always opens in a new tab." option checked. Should be a similar option in MAC.)

### Steps to reproduce

1. Current Container X (not default container)
2. Domain A **not assigned** to open in specific container
3. Click link to Domain A **outside FF** (e.g. in email client other than browser)
4. Result: Domain A link opens in DEFAULT Container

### Notes
Long-term problem, but sometime after Firefox Quantum

**Comment by terrizgit** [comment]
Anyone else had this problem?

Anyone from Mozilla see this?

**Comment by lhorace** [comment]
Hey there,

> Link is a redirect to a domain NOT marked as "always open in": opens (1 tab) in Default container no matter which container is open.

This seems like the expected behaviour if you opening a link outside of Firefox? It won't open in your currently open container. In fact, AFAIK, any link opened outside of Firefox always opens in a new tab. Though, there might be option somewhere to negate that behaviour, instead open in the current opened tab. Or are you suggesting when you open a link within a container (eg personal), it always opens in a new tab, Default container?

> Link clicked from email that redirects to domain NOT marked "always open in" opens in CURRENT container

FYI, there's allot open feature request wanting the opposite behaviour, if a Domain is not confined inside a container to always open in a Default container. AFAIK, this feature request hasn't implemented.  

> Link is a redirect to a domain marked as "always open in": opens 2 tabs in correct container

This sounds like a bug, however to isolate that it is MAC causing this, can you create a new FFx profile with just MAC installed and report back if the problem is reproducible ?

**Comment by terrizgit** [comment]
Hi lhorace - thanks for commenting :)

> any link opened outside of Firefox always opens in a new tab.

I agree it should open in a new **tab** (and it does). Thanks for that distinction - it's a link opened **outside Firefox**. I'll update my original ticket accordingly.

> are you suggesting when you open a link within a container (eg personal), it always opens in a new tab, Default container?

Not exactly. When I click a link **outside** FF, if it's a link (direct or redirected - I just confirmed that and will also update that above!) to a domain I don't have marked "always open in" it opens, in FF, in a new tab in the **Default** container. If it's a link I **have** marked "always open in Container X", it opens **two** tabs in Container X.

Context: I'm using Conex so I'm only looking at one container at a time. So opening a redirect link from an email changes the container I'm viewing, from Container X (the one I'd had open) to the Default container. If I want to see this new link AND see other links in Container X I was using, I have to right-click, open it in the container I was in, go back and close the tab in the Default container, and go back again to the Container X. Lots of steps for something that seems simple. Even if I wasn't using Conex, it seems cumbersome. Or something I'd like to be able to choose the behavior.

For example, if I wasn't using multi-account containers, and I clicked a link outside FF - it opens a new tab in the window I have active - it doesn't open a new window (because I have the "any link opened outside of Firefox always opens in a new tab." option checked. I just expect Multi-Containers to have the same kind of behavior/option.

I will try to reproduce with just MAC installed and report back...
